packages/connection/tests/php/test-rest-endpoints.php

```diff
@@ -76,21 +80,70 @@ public function tearDown() {
 		Constants::$set_constants['JETPACK__WPCOM_JSON_API_HOST'] = $this->api_host_original;
 
 		delete_transient( 'jetpack_assumed_site_creation_date' );
+
+		WorDBless_Options::init()->options = array();
 	}
 
 	/**
 	 * Testing the `/jetpack/v4/remote_authorize` endpoint.
 	 */
 	public function test_remote_authorize() {
+		add_filter( 'jetpack_options', array( $this, 'mock_jetpack_options' ), 10, 2 );
+		add_filter( 'pre_http_request', array( $this, 'intercept_auth_token_request' ), 10, 3 );
+
+		wp_cache_set(
+			self::USER_ID,
+			(object) array(
+				'ID'         => self::USER_ID,
+				'user_email' => 'sample@example.org',
+			),
+			'users'
+		);
+
+		$secret_1 = 'Az0g39toGWlYiTJ4NnDuAz0g39toGWlY';
+
+		$secrets = array(
+			'jetpack_authorize_' . self::USER_ID => array(
+				'secret_1' => $secret_1,
+				'secret_2' => 'zfIFcym2Jlzd8AVgzfIFcym2Jlzd8AVg',
+				'exp'      => time() + 60,
+			),
+		);
+
+		// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$options_filter = function( $value ) use ( $secrets ) {
+			return $secrets;
+		};
+		add_filter( 'pre_option_' . Manager::SECRETS_OPTION_NAME, $options_filter );
+
+		$user_caps_filter = function( $allcaps, $caps, $args, $user ) {
+			if ( $user instanceof WP_User && self::USER_ID === $user->ID ) {
+				$allcaps['manage_options'] = true;
+				$allcaps['administrator']  = true;
+			}
+
+			return $allcaps;
+		};
+		add_filter( 'user_has_cap', $user_caps_filter, 10, 4 );
+
 		$this->request = new WP_REST_Request( 'POST', '/jetpack/v4/remote_authorize' );
 		$this->request->set_header( 'Content-Type', 'application/json' );
-		$this->request->set_body( '{ "state": 111, "secret": "12345", "redirect_uri": "https://example.org", "code": "54321" }' );
+		$this->request->set_body( '{ "state": "' . self::USER_ID . '", "secret": "' . $secret_1 . '", "redirect_uri": "https://example.org", "code": "54321" }' );
 
 		$response = $this->server->dispatch( $this->request );
-		$data     = $response->get_data();
+		$this->assertEquals( 200, $response->get_status() );
 
-		$this->assertEquals( 404, $data['code'] );
-		$this->assertContains( '[user_unknown]', $data['message'] );
+		$data = $response->get_data();
+		$this->assertEquals( 'authorized', $data['result'] );
```

Comment:
What do you think of moving the 2 assert statements to the end of the test method, so that the cleanup stuff (removing filters, delete cache, reset user) will run even if an assert fails?